addons/mail/static/src/component_hooks/use_messaging_features/use_messaging_features.js

```diff
@@ -0,0 +1,18 @@
+/** @odoo-module */
+
+import { useModels } from "../use_models/use_models";
+import { useShouldUpdateBasedOnProps } from "../use_should_update_based_on_props/use_should_update_based_on_props";
+
+const { useComponent } = owl.hooks;
+
+export const componentRegistry = {};
+
+export function useMessagingFeatures(options) {
+    const component = useComponent();
+    useModels();
+    useShouldUpdateBasedOnProps(options);
+    const ComponentClass = component.constructor;
+    if (!componentRegistry.isPrototypeOf(ComponentClass.components)) {
```

Comment:
We might want to define another key to avoid this check (and I guess the need for `Object.create` below?) `extraComponents` maybe?

Reply:
Honestly, I think this is the perfect situation to use the prototype chain. I don't see why we should use a custom key instead. If you don't want to use prototypes, I'd rather use a WeakMap to track which classes have been processed than stuff data into the object itself.